src/components/accession2/Accession2Address.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import strings from 'src/strings';
+import { Link, Grid, Box, useTheme } from '@mui/material';
+import { AccessionPostRequestBody } from 'src/api/accessions2/accession';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import Textfield from 'src/components/common/Textfield/Textfield';
+import Autocomplete from 'src/components/common/Autocomplete';
+import { searchCountries } from 'src/api/country/country';
+import { getCountryByCode, getSubdivisionByCode } from 'src/utils/country';
+import { Country } from 'src/types/Country';
+
+type Accession2AddressProps = {
+  record: AccessionPostRequestBody;
+  onChange: (id: string, value?: any) => void;
+};
+
+export default function Accession2Address(props: Accession2AddressProps): JSX.Element {
+  const { record, onChange } = props;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [countries, setCountries] = useState<Country[]>();
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+
+  useEffect(() => {
+    const populateCountries = async () => {
+      const response = await searchCountries();
+      if (response) {
+        setCountries(response);
+      }
+    };
+    populateCountries();
+  }, []);
+
+  const gridSize = () => (isMobile ? 12 : 6);
+
+  const onChangeCountry = (newValue: string) => {
+    const found = countries?.find((country) => country.name === newValue);
+    if (found) {
+      onChange('collectionSiteCountryCode', found.code.toString());
+    }
+  };
+
+  const onChangeSubdivision = (newValue: string) => {
+    if (countries && record.collectionSiteCountryCode) {
+      const selectedCountry = getCountryByCode(countries, record.collectionSiteCountryCode);
```

Comment:
Could you use the getSelectedCountry here?

Reply:
good catch, i copy pasted from add new org.. will fix in both places.